core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -1265,6 +1275,66 @@ public static <T> void toHttp1Headers(HttpHeaders armeriaHeaders, T output,
         }
     }
 
+    /**
+     * Attempts to parse this URI's authority component and return {@link URI}.
+     * The authority part may have one of the following formats (The userinfo part will be ignored.):
+     * <ul>
+     *   <li>{@code "<host>:<port>"} for a host endpoint </li>
+     *   <li>{@code "<host>"}, {@code "<host>:"} for a host endpoint with no port number specified</li>
+     * </ul>
+     * An IPv4 or IPv6 address can be specified in lieu of a host name, e.g. {@code "127.0.0.1:8080"} and
+     * {@code "[::1]:8080"}.
+     *
+     * @throws IllegalArgumentException if {@code authority} do not comply with RFC 2396
+     */
+    public static URI normalizeAuthority(String authority) {
+        requireNonNull(authority, "authority");
+        final String authorityWithoutUserInfo = removeUserInfo(authority);
+        try {
+            final URI uri = new URI(null, authorityWithoutUserInfo, null, null, null);
+            return uri.parseServerAuthority();
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Attempts to parse this URI's scheme and authority component and return {@link URI}.
+     * The authority part may have one of the following formats (The userinfo part will be ignored.):
+     * <ul>
+     *   <li>{@code "<host>:<port>"} for a host endpoint </li>
+     *   <li>{@code "<host>"}, {@code "<host>:"} for a host endpoint with no port number specified</li>
+     * </ul>
+     * An IPv4 or IPv6 address can be specified in lieu of a host name, e.g. {@code "127.0.0.1:8080"} and
+     * {@code "[::1]:8080"}.
+     *
+     * @throws IllegalArgumentException if {@code scheme} or {@code authority} do not comply with RFC 2396
+     */
+    public static URI normalizeSchemeAndAuthority(String scheme, String authority) {
+        requireNonNull(scheme, "scheme");
+        requireNonNull(authority, "authority");
+
+        if (!SCHEME_VALIDATOR.test(scheme)) {
+            throw new IllegalArgumentException("scheme: " + scheme + " (expected: a valid scheme)");
+        }
+
+        final String authorityWithoutUserInfo = removeUserInfo(authority);
+        try {
+            final URI uri = new URI(scheme, authorityWithoutUserInfo, null, null, null);
+            return uri.parseServerAuthority();
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
```

Comment:
Is it possible to just delegate to `normalizeAuthority` instead?

```suggestion
        return normalizeAuthority(authority);
```

Reply:
I think that's good!!